### PR TITLE
Fix/path traversal

### LINKED ISSUE #104: bug/validate-input
The HTTP request URL and the filename of POST upload are susceptible to [path traversal](https://owasp.org/www-community/attacks/Path_Traversal) exploits.

Example:
```shell
python
>>> import requests
>>> r = requests.get("http://localhost:8080/../../../../../../../../../../../../../../etc/passwd")
>>> print(r.text)
```

Possible fix:
Either disallow `/..` in the request URL and the filename of POST upload or simplify the paths.

### PR DESCRIPTION
**Pull request by gbudau**
resolve #104 

This is a stacked/incremental PR. It depends on #108, and we should merge it after we merge #108.

### CODE REVIEW

File: srcs/app/http/HttpRequest.cpp [hunk 1]
```diff
@@ -206,6 +206,10 @@ bool	HttpRequest::IsValidPath_(const std::string &path) const {
 	return !path.empty() && path[0] == '/';
 }
 
+bool	HttpRequest::IsValidDecodedPath_(const std::string &path) const {
```
Comment (Damrod) [inline srcs/app/http/HttpRequest.cpp:209]: I already mentioned that I don't mind this much but if we're going to do, we may as well be strict about it.

We don't accept "/..", but what about "../"?

**Review by Damrod — APPROVED** [review]
Very good. Thank you!

File: srcs/app/http/HttpRequest.cpp (hunk 1 above)
Comment (gbudau) [reply srcs/app/http/HttpRequest.cpp:209]: Great point.
Also, the path could be a filename starting with "..",
for example, "/.....name", and this should be valid.

Comment (gbudau) [reply srcs/app/http/HttpRequest.cpp:209]: Great point! Also, the path could be a filename starting with "..",
for example, "/.....name", and this should be valid.

Comment (gbudau) [reply srcs/app/http/HttpRequest.cpp:209]: @Damrod, the last commit should fix these cases.

**Review by vdedios — APPROVED** [review]
LGTM!

**Comment by gbudau** [comment]
Thank you! :tada:

**Pull request merged**